packages/core/src/provider/base.ts

```diff
@@ -293,8 +293,8 @@ export class SmartAccountProvider<
     const initCode = await this.account.getInitCode();
     const uoStruct = await asyncPipe(
       this.dummyPaymasterDataMiddleware,
-      this.gasEstimator,
       this.feeDataGetter,
+      this.gasEstimator,
```

Comment:
@moldy530 can you check my logic on this change. We want the fees to get estimated and applied before gas is estimated as gas estimation can sometimes rely on fees (e.g. as it will on Optimism). Does this have that intended impact?

Reply:
yup! pipe executes sequentially so this will do fees then gas